scrape/scrape.go

```diff
@@ -1654,7 +1653,7 @@ loop:
 				// Bypass staleness logic if there is an explicit timestamp.
 				sl.cache.trackStaleness(hash, lset)
 			}
-			sl.cache.addRef(mets, ref, lset, hash)
+			sl.cache.addRef(met, ref, lset, hash)
```

Comment:
Just curious, are `mets` and `met` the same thing? Or will using `met` instead of `mets` still achieve the desired effect?

`mets` looks like it's populated from the `p.Metric` function only when `ok` is false:

https://github.com/prometheus/prometheus/blob/bec5abc4dc63831dfbfba50bc3230a6fea6b78e0/scrape/scrape.go#L1597-L1604

while `met` seems to be always populated using the `p.Series`/`p.Histogram` function:

https://github.com/prometheus/prometheus/blob/bec5abc4dc63831dfbfba50bc3230a6fea6b78e0/scrape/scrape.go#L1569-L1574

Reply:
`met` is a `[]byte` containing the metric being parsed, for example `foo{x="bar}"`.
`mets` is a string copy of that, populated only by `p.Metric`, which is only called if we did not find the text in the cache, indicated by `ok` being false.

Since, on this branch, `mets` is only used to call `addDropped`, I think I will just remove it.

Reply:
Ok, sounds good 👍🏻